src/components/DocumentCard/DocumentCard.tsx

```diff
@@ -31,16 +27,30 @@ export class DocumentCard extends React.Component<IDocumentCardProps, any> {
           css(
             'ms-DocumentCard',
             {
-              'ms-DocumentCard--actionable': onClick ? true : false,
+              'ms-DocumentCard--actionable': actionable,
               'ms-DocumentCard--compact': type === DocumentCardType.compact ? true : false
             },
             className
           )
         }
-        onClick={ onClick }
+        onClick={ actionable ? this._onClick : null }
```

Comment:
DocumentCard is different than DocumentCardLocation. It is a Div with children components but not an <a> tag element. We want to make sure when user only set onClickHref but not pass in onClick handler, click on the card can redirect to the corresponding url.